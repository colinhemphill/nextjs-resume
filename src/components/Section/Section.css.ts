import { styleVariants } from '@vanilla-extract/css';
import { atoms } from '../../strum-design-system/sprinkles.css';

const sectionBaseStyle = atoms({
  paddingY: { xs: 6, md: 6, lg: 8 },
});

export const sectionStyle = styleVariants({
  alternate: [
    sectionBaseStyle,
    atoms({
      backgroundColor: { darkMode: 'medium', lightMode: 'light' },
      color: { darkMode: 'white', lightMode: 'dark' },
    }),
  ],
  standard: [
    sectionBaseStyle,
    atoms({
      backgroundColor: { darkMode: 'dark', lightMode: 'white' },
      color: { darkMode: 'white', lightMode: 'dark' },
    }),
  ],
});
