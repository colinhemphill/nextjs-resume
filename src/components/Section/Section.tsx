import React, { PropsWithChildren } from 'react';
import Container from '../../strum-design-system/components/Container/Container';
import { sectionStyle } from './Section.css';

interface Section {
  color?: keyof typeof sectionStyle;
}

const Section: React.FC<PropsWithChildren<Section>> = (props) => {
  const { children, color } = props;

  return (
    <section className={sectionStyle[color]}>
      <Container>{children}</Container>
    </section>
  );
};

export default Section;
