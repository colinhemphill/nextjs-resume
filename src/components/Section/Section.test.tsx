import React from 'react';
import { render } from '../../test/testUtils';
import Section from './Section';

jest.mock('next/router', () => require('next-router-mock'));

describe('<Section />', () => {
  beforeEach(() => {
    jest.resetModules();
  });

  it('Matches the snapshot', () => {
    const { container } = render(
      <Section color="alternate">
        <div>Test</div>
      </Section>,
    );
    expect(container).toMatchSnapshot();
  });
});
