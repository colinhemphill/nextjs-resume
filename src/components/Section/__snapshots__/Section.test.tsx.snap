// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> Matches the snapshot 1`] = `
<div>
  <section
    class="Section_sectionStyle_alternate__1t64bv43 Section__1t64bv40 sprinkles_paddingTop_6_xs__1pzb2zolo sprinkles_paddingTop_6_md__1pzb2zolq sprinkles_paddingTop_8_lg__1pzb2zom3 sprinkles_paddingBottom_6_xs__1pzb2zof6 sprinkles_paddingBottom_6_md__1pzb2zof8 sprinkles_paddingBottom_8_lg__1pzb2zofl Section__1t64bv41 sprinkles_backgroundColor_medium_darkMode__1pzb2zono sprinkles_backgroundColor_light_lightMode__1pzb2zonn sprinkles_color_white_darkMode__1pzb2zopf sprinkles_color_dark_lightMode__1pzb2zooz"
  >
    <div
      class="Container_containerStyle__rjkhb02 Container_containerFluidStyle__rjkhb01 margin_mxAuto__1nlkou20 Container__rjkhb00 sprinkles_paddingLeft_4_xs__1pzb2zoh0 sprinkles_paddingRight_4_xs__1pzb2zoj6"
    >
      <div>
        Test
      </div>
    </div>
  </section>
</div>
`;
